src/Bicep.Deploy/DeploymentManager.cs

```diff
@@ -19,27 +20,69 @@ public async Task<ArmDeploymentResource> CreateOrUpdateAsync(ArmDeploymentDefini
         {
             var armClient = new ArmClient(new DefaultAzureCredential());
 
-            if (deploymentDefinition.SubscriptionId is { } subscriptionId &&
-                subscriptionId.Equals(Guid.Empty.ToString(), StringComparison.OrdinalIgnoreCase))
+            try
             {
-                var subscription = await armClient.GetDefaultSubscriptionAsync(cancellationToken);
-                deploymentDefinition = deploymentDefinition with { SubscriptionId = subscription.Id };
-            }
+                if (deploymentDefinition.SubscriptionId is { } subscriptionId &&
+                subscriptionId.Equals(Guid.Empty.ToString(), StringComparison.OrdinalIgnoreCase))
+                {
+                    var subscription = await armClient.GetDefaultSubscriptionAsync(cancellationToken);
+                    deploymentDefinition = deploymentDefinition with { SubscriptionId = subscription.Id };
+                }
 
-            var deploymentResource = armClient.GetArmDeploymentResource(deploymentDefinition.Id);
-            var deploymentContent = new ArmDeploymentContent(deploymentDefinition.Properties);
+                var deploymentResource = armClient.GetArmDeploymentResource(deploymentDefinition.Id);
+                var deploymentContent = new ArmDeploymentContent(deploymentDefinition.Properties);
 
-            var operation = await deploymentResource.UpdateAsync(WaitUntil.Completed, deploymentContent, cancellationToken);
+                var operation = await deploymentResource.UpdateAsync(WaitUntil.Completed, deploymentContent, cancellationToken);
 
-            return operation.Value;
+                return operation.Value;
+            }
+            catch (RequestFailedException ex)
+            {
+                throw new DeploymentException(ex.Message, ex);
+            }
         }
 
-        public Task<ArmDeploymentResource> CreateOrUpdateAsync(ArmDeploymentDefinition deploymentDefinition, Action<ImmutableSortedSet<ArmDeploymentOperation>>? onOperationsUpdated, CancellationToken cancellationToken)
+        public async Task<ArmDeploymentResource> CreateOrUpdateAsync(ArmDeploymentDefinition deploymentDefinition, Action<ImmutableSortedSet<ArmDeploymentOperation>>? onOperationsUpdated, CancellationToken cancellationToken)
         {
             // TODO: Custom polling logic. Take a look at:
             // - https://github.com/Azure/azure-powershell/blob/b819adaeec70e735e3b46d51c72bc2fc17c4fef6/src/Resources/ResourceManager/SdkClient/NewResourceManagerSdkClient.cs#L201
             // - Azure.Core.OperationPoller
-            throw new NotImplementedException();
+            //throw new NotImplementedException();
+            var armClient = new ArmClient(new DefaultAzureCredential());
+
+            try
+            {
+                if (deploymentDefinition.SubscriptionId is { } subscriptionId &&
+                    subscriptionId.Equals(Guid.Empty.ToString(), StringComparison.OrdinalIgnoreCase))
```

Comment:
Out of interest, why is this check necessary?

Reply:
not sure tbh, @shenglol ?

Reply:
The empty GUID serves as a placeholder to prevent build failure when `subscription()` is used but the user hasn't logged in before running the build command.

```javascript
deploy 'main.bicep' {
  scope: subscription() // No explicit ID set. Gets compiled to an empty GUID.
  ...
}
```

I think it may make sense to defer resolving the subscription until deployment time.

Reply:
We may also need to apply the same logic to `tenant()`.